### ISSUE TITLE
Can't get custom ContractResolver to work

### ISSUE THREAD

**Issue by IvanSorokin** [elastic/elasticsearch-net#3494]
<!--
GitHub is reserved for bug reports and feature requests. The best place
to ask a general question is at the Elastic Discourse forums at
https://discuss.elastic.co. If you are in fact posting a bug report or
a feature request, please include one and only one of the below blocks
in your new issue.
-->

<!--
If you are filing a bug report, please remove the below feature
request block and provide responses for all of the below items.
-->

**NEST/Elasticsearch.Net version**: 6.4.0

**Elasticsearch version**: 6.2.2

**Description**: I want to use $type only on root object. I tried to use ContractResolver from this
[answer](https://stackoverflow.com/questions/36356336/json-net-how-to-add-property-type-only-on-root-object/36361605#36361605) but it didn't work. Then I tried to modify this resolver, even threw exceptions from it but it didn't change anything. Seems like it is not used at all.

**Client creation**:
```
                         ConnectionSettings.SourceSerializerFactory sourceSerializerFactory = (builtin, setts) => new JsonNetSerializer(
                                                                                                                     builtin, setts,
                                                                                                                     () => new JsonSerializerSettings
                                                                                                                           {
                                                                                                                               TypeNameHandling = TypeNameHandling.All,
                                                                                                                               NullValueHandling = NullValueHandling.Ignore,
                                                                                                                               ContractResolver = SuppressItemTypeNameContractResolver.Instance,
                                                                                                                               TypeNameAssemblyFormatHandling = TypeNameAssemblyFormatHandling.Simple,
                                                                                                                               
                                                                                                                           }
                                                                                                                    );
                         var connectionSettings =
                             new ConnectionSettings(connectionPool, sourceSerializer: sourceSerializerFactory);
```

**Resolver**:

```
public class SuppressItemTypeNameContractResolver : DefaultContractResolver
        {
            static SuppressItemTypeNameContractResolver()
            {
                Instance = new SuppressItemTypeNameContractResolver();
            }

            public static SuppressItemTypeNameContractResolver Instance { get; }

            protected override JsonContract CreateContract(Type objectType)
            {
                var contract = base.CreateContract(objectType);

                if (contract is JsonContainerContract containerContract)
                {
                    if (containerContract.ItemTypeNameHandling == null)
                        containerContract.ItemTypeNameHandling = TypeNameHandling.None;
                }

                return contract;
            }
        }
```

**Comment by IvanSorokin** [comment]
okay, after inspecting source code I can assume that this option really doesn't work because it is always overrided by elastic resolver afterwards. a bit misleading for me that you can pass an argument but have no changes in the final result.

**Comment by russcam** [comment]
I've opened #3599 for discussion

**Comment by russcam** [comment]
#3599 has been merged and will be in the next release (6.6.0), so am closing this issue.
